scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java

```diff
@@ -0,0 +1,145 @@
+import org.ow2.proactive.scheduler.task.context.TaskContext;
+import org.ow2.proactive.scheduler.task.context.TaskContextVariableExtractor;
+import org.ow2.proactive.scheduler.task.executors.forked.env.ForkedJvmTaskExecutionCommandCreator;
+import org.ow2.proactive.scheduler.task.utils.ForkerUtils;
+import org.ow2.proactive.scripting.ScriptResult;
+
+public class ForkedProcessBuilderCreator implements Serializable {
+    private static final Set<PosixFilePermission> SHARED_FOLDER_PERMISSIONS = PosixFilePermissions.fromString(
+            "rwxrwxrwx");
+    private final ForkedJvmTaskExecutionCommandCreator forkedJvmTaskExecutionCommandCreator = new ForkedJvmTaskExecutionCommandCreator();
+    private final TaskContextVariableExtractor taskContextVariableExtractor = new TaskContextVariableExtractor();
+    private final ForkEnvironmentScriptExecutor forkEnvironmentScriptExecutor = new ForkEnvironmentScriptExecutor();
+
+    /**
+     * Creates a process builder for a given task context.
+     *
+     * @param context           The task context to execute.
+     * @param serializedContext The task context saved to disk.
+     * @param outputSink        Standard output sink.
+     * @param errorSink         Error sink.
+     * @param workingDir        The working directory to execute the process in.
+     * @return Returns a process builder, ready to excute.
+     * @throws Exception
+     */
+    public OSProcessBuilder createForkedProcessBuilder(TaskContext context, File serializedContext,
+            PrintStream outputSink, PrintStream errorSink, File workingDir) throws Exception {
+
+        String nativeScriptPath = context.getSchedulerHome();
+        ScriptResult forkEnvironmentScriptResult = null;
+
+        OSProcessBuilder processBuilder = getOsProcessBuilder(context, workingDir, nativeScriptPath);
+
+        forkEnvironmentScriptResult = executeForkEnvironmentScriptAndExtractVariables(context, outputSink,
+                errorSink, forkEnvironmentScriptResult,
+                processBuilder);
+
+        processBuilder
+                .command()
+                .addAll(forkedJvmTaskExecutionCommandCreator
+                        .createForkedJvmTaskExecutionCommand(context,
+                                forkEnvironmentScriptResult, serializedContext.getAbsolutePath()
+                        ));
+
+        processBuilder = processBuilder.directory(workingDir);
+        return processBuilder;
+    }
+
+    private ScriptResult executeForkEnvironmentScriptAndExtractVariables(TaskContext context,
+            PrintStream outputSink, PrintStream errorSink, ScriptResult forkEnvironmentScriptResult,
+            OSProcessBuilder processBuilder) throws Exception {
+        ForkEnvironment forkEnvironment = context.getInitializer().getForkEnvironment();
+        if (forkEnvironment != null) {
+
+            if (forkEnvironment.getEnvScript() != null) {
+                forkEnvironmentScriptResult = forkEnvironmentScriptExecutor
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "forkEnvironmentScriptResult". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)